framework/src/st_things/things_stack/utils/things_network.c

```diff
@@ -254,7 +254,8 @@ int things_get_ap_list(access_point_info_s** p_info, int* p_count)
 			snprintf(pinfo->e_ssid, WIFIMGR_SSID_LEN, "%s", wifi_scan_iter->e_ssid);
 			snprintf(pinfo->bss_id, WIFIMGR_MACADDR_STR_LEN, "%s", wifi_scan_iter->bss_id);
 			snprintf(pinfo->signal_level, MAX_LEVEL_SIGNAL, "%d", wifi_scan_iter->signal_level);
-
+			snprintf(pinfo->auth_type, MAX_TYPE_AUTH, "%s", wifi_scan_iter->auth_type);
```

Comment:
'MAX_TYPE_AUTH - 1' or
'char auth_type[MAX_TYPE_AUTH + 1];'	in struct access_point_info_s